### ISSUE TITLE
K/D requirements are Diverting players to just play for kills and not Capture The Flag

### ISSUE THREAD

**Issue by moreavy** [MT-CTF/capturetheflag#809]
This is discussion to have a solution for players not capturing and just playing for kills.

**Comment by Minetest-j45** [comment]
see https://github.com/MT-CTF/capturetheflag/blob/master/mods/ctf/ctf_stats/init.lua line 210
```lua
	return stats.score >= 10000 and kd >= 1.5
``` just remove the "and kd >= 1.5".

**Comment by moreavy** [comment]
> see https://github.com/MT-CTF/capturetheflag/blob/master/mods/ctf/ctf_stats/init.lua line 210
> 
> ```lua
> 	return stats.score >= 10000 and kd >= 1.5``` 
> just remove the "and kd >= 1.5".
> ```

We can't call that a solution for this issue.

**Comment by Minetest-j45** [comment]
it will remove the need for kd for prochest, so people trying to get prochest wont need kd and will try get the points needed by capturing the flag.

**Comment by Anthony-De** [comment]
Instead of removing the kd requirement adding a kd independent qualification might be better.

Maybe something like this:
```lua
return stats.score >= 10000 and kd >= 1.5 or stats.score >= <insert_point_limit>
```
Thoughts?

**Comment by moreavy** [comment]
```lua
function ctf_stats.is_pro(name)
	local stats = ctf_stats.player(name)
	local kd = stats.kills / (stats.deaths == 0 and 1 or stats.deaths)
        local ca = stats.captures / stats.attempts
	return (stats.score >= 10000 and kd >= 1.5) or (ca >= 0.3 and stats.captures >= 20)
end
```

The requirements of k/d will remain the same here, but people good in capturing and less good in PvP will get pro section too.

**Comment by moreavy** [comment]
Now, Capture are pro requirements. But yet, if players make 10 captures and then just going for k/d is also a problem.

**Comment by savilli** [comment]
Better to do captures / kills > 0.005. If you kill a lot you should capture

**Comment by realOneplustwo** [comment]
Maybe captures per match > 0.05. In other words, you should capture at least once every 20 matches you participate in.

**Comment by appgurueu** [comment]
> Maybe captures per match > 0.05. In other words, you should capture at least once every 20 matches you participate in.

That would require the addition of a "match" stat and would accordingly be breaking.

**Comment by moreavy** [comment]
### Experience
People who played the game for a long time and know the game, would be eligible for pro section. As @Anthony-De said, a certain amount of score per the time the players has played will determine he/she if pro or not. We can add that stat soon, and I think it will be a good idea to influence players the game and not just camping for kills. Also a certain amount of Captures and 1.o k/d should be a requirement.

**Comment by appgurueu** [comment]
@savilli reason for the closing of this discussion? AFAIK CTFv3 hasn't changed much in that regard.

**Comment by LoneWolfHT** [comment]
Issues can be reopened if they still apply and support is shown

**Comment by LoneWolfHT** [comment]
I'm going to look into counting kill assists for this

**Comment by mieu12** [comment]
I think kills should still be counted after 1-2 seconds after a hit is dealt on the enemy instead of just kill assists, only after the 1-2 seconds it would count as a kill assists.

**Comment by farooqkz** [comment]
> I think kills should still be counted after 1-2 seconds after a hit is dealt on the enemy instead of just kill assists, only after the 1-2 seconds it would count as a kill assists.

I disagree. As a sniper in game, sometimes I need 10-20 seconds to kill someone or to damage someone so that a teammate will kill them easily.

**Comment by magnetar47** [comment]
I agree that score and KD alone won't be a good enough measurement of one's "pro-ness". KD can vary massively according to different playstyles. One can have a "good" KD if they employ a playstyle that's heavily dependent on camping. One can have "bad" KD if they're more upfront in their combat style and keep charging. Players who relentlessly keep trying to capture the flag will by design have terrible KDs but that doesn't mean they're not pro.

My suggestion is that instead of having a one-rule-fits-all approach, we can have multiple criteria, out of which, if even one is satisfied, that player is considered a "pro". Such an approach won't force players to play in a certain way just to become a pro, and would instead encourage each player to employ their own preferred playstyles. Examples of such criteria can be:

- Flat score + Score/Deaths (for people with a more PvP-oriented playstyle)
- Flat score + Captures/Deaths (for people who focus on capturing the flag)
- Flat score + Total HP healed/Deaths (for people who prefer to be a medic)
- etc.

IMO, it's good to also have a flat score requirement (5000 may be a good requirement?), otherwise one can be a pro after 30 score and 0 deaths, for example.

Please note that I haven't given too much thought to these example criteria, so don't critique me on this. :)

**Comment by farooqkz** [comment]
> I agree that score and KD alone won't be a good enough measurement of one's "pro-ness". KD can vary massively according to different playstyles. One can have a "good" KD if they employ a playstyle that's heavily dependent on camping. One can have "bad" KD if they're more upfront in their combat style and keep charging. Players who relentlessly keep trying to capture the flag will by design have terrible KDs but that doesn't mean they're not pro.
> 
> My suggestion is that instead of having a one-rule-fits-all approach, we can have multiple criteria, out of which, if even one is satisfied, that player is considered a "pro". Such an approach won't force players to play in a certain way just to become a pro, and would instead encourage each player to employ their own preferred playstyles. Examples of such criteria can be:
> 
>     * Flat score + Score/Deaths (for people with a more PvP-oriented playstyle)
> 
>     * Flat score + Captures/Deaths (for people who focus on capturing the flag)
> 
>     * Flat score + Total HP healed/Deaths (for people who prefer to be a medic)
> 
>     * etc.
> 
> 
> IMO, it's good to also have a flat score requirement (5000 may be a good requirement?), otherwise one can be a pro after 30 score and 0 deaths, for example.
> 
> Please note that I haven't given too much thought to these example criteria, so don't critique me on this. :)

Good ideas. However a reminder: Pro is the one who doesn't waste resources in the teamchest if you give them access to the extra section.

**Comment by farooqkz** [comment]
> ### Experience
> 
> People who played the game for a long time and know the game, would be eligible for pro section. As @Anthony-De said, a certain amount of score per the time the players has played will determine he/she if pro or not. We can add that stat soon, and I think it will be a good idea to influence players the game and not just camping for kills. Also a certain amount of Captures and 1.o k/d should be a requirement.

I was thinking of this idea last night and today I see someone else had the idea before me: `score/time_online`

And I think it will cover what @magnetar47 has in mind as well. Because we'll be counting healings, kill assists and actual kills all together. Two may try to kill savilli, one causes considerable damage to him and the other kills him. Both get a considerable amount of score because savilli is a big fish. On the other hand, campers won't benefit much from this because pros are hard to kill even if you surprise them.

However there are many things to consider: 

 - There are other ways which people benefit their team but they don't get score for them. Like bringing weapons from battlefield back to teamchest or building defense structures.
 - Must score from bounties be counted or not? With the current algorithm for calculating bounty score, one can get a very considerable score by killing someone. The one who has assisted will get only assist points while the killer gets 50(for example).

**Comment by farooqkz** [comment]
@LoneWolfHT opinion?

**Comment by farooqkz** [comment]
I want to make a PR based on a new definition of pro. Anyone with at least one of these requirements will be considered a pro:

 - At least 14 kills per 10 deaths(each 3 "half" kill assists will count as 1 kill)
 - At least 1 capture per 14 deaths
 - At least 30 HP of healing per 1 death

**Comment by LoneWolfHT** [comment]
> At least 1 capture per 14 deaths
This is unfair, unless those are three different ways to get pro chest access

**Comment by farooqkz** [comment]
Yes that's true. One must have at least one of these requirements to be considered pro. So one can focus only on capturing without having to worry about their K/D and like this about other things.

**Comment by farooqkz** [comment]
I think 12.5 kills per 10 deaths is more sensible for made fight as two parties killing each other with bases is common.
